pkg/controller/worker/genericactuator/interface.go

```diff
@@ -43,6 +42,11 @@ type WorkerDelegate interface {
 
 	// GenerateMachineDeployments generates the configuration for the desired machine deployments.
 	GenerateMachineDeployments(context.Context) (worker.MachineDeployments, error)
+
+	// GetMachineImages returns the list of used machine images for this `Worker` resource. It will be stored in the
+	// `.status.providerStatus` field of the `Worker` resource such that the controller can look up its provider-specific
+	// machine image information in case the required version has been removed from its componentconfig.
+	GetMachineImages(context.Context) (runtime.Object, error)
```

Comment:
Basically, I would follow to the `infrastructure` controllers and call it `GetWorkerConfig` but I'm also fine to call it `GetMachineImages` now, since we only have this special case. Your decision ;)